src/Process/LayeredMaterialNodeProcessing.js

```diff
@@ -0,0 +1,309 @@
+    if (!force) {
+        // does this tile needs a new texture?
+        if (!node.isColorLayerDownscaled(layer.id)) {
+            return;
+        }
+        // is fetching data from this layer disabled?
+        // TODO
+        /*
+        if (!layersConfig.isColorLayerVisible(layer.id) ||
+            layersConfig.isLayerFrozen(layer.id)) {
+            return;
+        }
+        */
+    }
+
+    let ancestor = null;
+
+    const currentLevel = node.materials[RendererConstant.FINAL].getColorLayerLevelById(layer.id);
+    // if this tile has no texture (level == -1), try use one from an ancestor
+    if (currentLevel === -1) {
+        ancestor = findAncestorWithValidTextureForLayer(node, parent || node.parent, l_COLOR, layer);
+    } else {
+        var targetLevel = chooseNextLevelToFetch(layer.updateStrategy.type, node.level, currentLevel, layer.updateStrategy.options);
+        if (targetLevel === currentLevel) {
+            return;
+        }
+        if (targetLevel < node.level) {
+            ancestor = node.getNodeAtLevel(targetLevel);
+        }
+    }
+
+    node.layerUpdateState[layer.id].newTry();
+
+    const command = {
+        /* mandatory */
+        layer,
+        requester: node,
+        priority: nodeCommandQueuePriorityFunction(node),
+        earlyDropFunction: force ? null : refinementCommandCancellationFn,
+        /* specific params */
+        ancestor,
+        redraw: !force,
+    };
+
+    return context.scheduler.execute(command).then(
+        (result) => {
+            if (node.material === null) {
+                return;
+            }
+            if (!result) {
+                // shouldn't happen
+                // eslint-disable-next-line no-console
+                console.warn(`Invalid value received. Probably a bug in the provider (${layer.protocol})`);
+                return;
+            }
+
+            const level = ancestor ? ancestor.level : node.level;
+            // Assign .level to texture
+            if (Array.isArray(result)) {
+                for (let j = 0; j < result.length; j++) {
+                    result[j].texture.level = level;
+                }
+
+                node.setTexturesLayer(result, l_COLOR, layer.id);
+            } else if (result.texture) {
+                result.texture.level = level;
+                node.setTexturesLayer([result], l_COLOR, layer.id);
+            }
+
+            node.layerUpdateState[layer.id].success();
+
+            return result;
+        },
+        (err) => {
+            if (err instanceof CancelledCommandException) {
+                node.layerUpdateState[layer.id].success();
+            } else {
+                node.layerUpdateState[layer.id].failure(Date.now());
+                // TODO event
+                context.scene.notifyChange(node.layerUpdateState[layer.id].secondsUntilNextTry() * 1000, false);
+            }
+        });
+}
+
+export function updateLayeredMaterialNodeElevation(context, layer, node) {
```

Comment:
Well.
The private function returns a promise (which is used in `initNewNode`) but the regular update function should return nothing or an array of elements to update...
I agree this is rather ugly, but it gets the job done for now. Maybe a good candidate for refactoring later though.